src/platform/monitoring/DowntimeNotification/actions/index.js

```diff
@@ -0,0 +1,23 @@
+import { apiRequest } from '../../../utilities/api';
+import { createServiceMap } from '../util/helpers';
+
+export const RETREIVE_SCHEDULED_DOWNTIME = 'RETREIVE_SCHEDULED_DOWNTIME';
+export const RECEIVE_SCHEDULED_DOWNTIME = 'RECEIVE_SCHEDULED_DOWNTIME';
+
+export function getScheduledDowntime() {
+  return async (dispatch) => {
+    dispatch({ type: RETREIVE_SCHEDULED_DOWNTIME });
+
+    let response;
+    try {
+      response = await apiRequest('/maintenance_windows/');
+    } catch (err) {
+      // Probably in a test environment and the route isn't mocked.
+    } finally {
+      dispatch({
+        type: RECEIVE_SCHEDULED_DOWNTIME,
+        map: createServiceMap(response && response.data)
```

Comment:
A maintainer is going to look at the reducer of this action to find its behavior. Using a helper is fine but it's probably easier to track down a behavior if this is composed in the reducer. Putting this in the action muddles the purpose of an action vs reducer- what criteria do we use to decide if this should be composed here or in the reducer?

Reply:
That's raises an interesting question, because I tend to use actions as the starting point for understanding how data flows throughout the store. Mapping raw API data is simplest to me conceptually within the context of the request issuer, but either way, the work is abstracted into the helper so I can don't really feel strongly.

@jbalboni @cvalarida Do you all have thoughts on @rianfowler's point?

Reply:
I generally try to put as much logic as possible into reducers, rather than action creators.

Reply:
That's interesting, because it seems like we do a lot of property mapping directly after an API call, such as in the user `getProfile` action. I'll revisit this.

Reply:
That's my opinion, we are not consistent about it at all. Would be good to have some general guidelines for us as a group.

Reply:
Also, I mean "as possible" as in, within the bounds of Redux's guidelines, which is that reducers should be be pure functions that only depend on the previous state and action object and do not mutate data.

Reply:
I don't think it's a big concern either way in this case since that action is dispatched at most one time and the logic is all contained in a helper, but I'm definitely open to moving to the reducer approach. I'll have to adjust my test cases though so I'll have to circle back around.

Reply:
The other approach I've seen is to categorize this kind of work- mapping one schema to another- as a `parser` and to run API responses through a `parser` before they enter redux if the shape of the data in the response does not match the client state. This refines the purpose of redux as strictly business logic and behavior.

Reply:
The important thing to me is that the patterns of work are organized consistently. I don't really care where they go. 

We should have a larger discussion about domain language and the kind of composition we want to encourage in PRs. I usually prioritize reducing cognitive load for maintainers above everything and I know that's not the best opinion in all circumstances.